contrib/postgresql/template.py

```diff
@@ -0,0 +1,106 @@
+pkgname = "postgresql"
+pkgver = "16.2"
+pkgrel = 0
+build_style = "configure"
+configure_args = [
+    # TODO: review path choices - alpine puts pkgname in a few of these
+    "--bindir=/usr/bin",
+    "--libdir=/usr/lib",
+    "--includedir=/usr/include",
+    "--datadir=/usr/share",
+    "--disable-rpath",  # TODO: alpine does this but the docs say not to. should we?
+    "--with-system-tzdata=/usr/share/zoneinfo",
+]
+make_cmd = "gmake"
+makedepends = [
+    "gmake",
+    "pkgconf",
+    "icu-devel",
+    "readline-devel",
+    "zlib-devel",
+    "linux-headers",
+]
+depends = ["tzdata"]
+pkgdesc = "Sophisticated object-relational DBMS"
+maintainer = "mia <mia@mia.jetzt>"
+license = "PostgreSQL"
+url = "https://www.postgresql.org"
+source = [
+    f"https://ftp.postgresql.org/pub/source/v{pkgver}/postgresql-{pkgver}.tar.bz2"
+]
+sha256 = ["446e88294dbc2c9085ab4b7061a646fa604b4bec03521d5ea671c2e5ad9b2952"]
+# TODO: checks rely on libpq already being installed
+# could maybe makedepend on libpq once it's in the repos?
+# note: alpine doesn't do checks either
+options = ["!check"]
+
+
+# NOTE: most of these subpackages are based on whatever alpine is doing, so they might need tweaking
+
+
+@subpackage("libpq")
+def _libpq(self):
+    self.pkgdesc = "PostgreSQL client library"
+    return ["usr/lib/libpq.so*"]
+
+
+@subpackage("libpq-devel")
+def _libpq_devel(self):
+    self.depends += ["libpq"]
+    return [
+        "usr/bin/pg_config",
+        "usr/include/libpq-*.h",
+        "usr/include/libpq/*",
+        "usr/include/pg_config*.h",
+        "usr/include/postgres_ext.h",
+        "usr/lib/libpq.*",
+        "usr/lib/libpgcommon*.a",
+        "usr/lib/libpgport*.a",
+        "usr/lib/pkgconfig/libpq.pc",
+        # TODO: should this be included here?
+        # they aren't on alpine, but i don't see why postgres would need them
+        # either way, the file triggers static library lints
+        "usr/lib/libpgfeutils.a",
+    ]
+
+
+@subpackage("libecpg")
+def _libecpg(self):
+    self.pkgdesc = "Embedded SQL in C"
+    return [
+        "usr/lib/libecpg.so*",
+        "usr/lib/libpgtypes.so*",  # TODO: should libpgtypes be it's own package?
```

Comment:
alternate question- what benefit do we get from splitting the two libs separately, instead of just one `postgresql-libs`?

Reply:
some postgresql libraries seem to not be tied to specific postgresql versions. the alpine libpq package is pulled from the latest postgres. maybe this would be a good argument for it?
https://git.alpinelinux.org/aports/tree/main/postgresql16/APKBUILD#n55

Reply:
additionally, we could split out a `postgresql-libs` and a `postgresql16`/`postgresql17`

Reply:
yeah, that was the next part. if everything is fine using latest libpg/ecpg/pgtypes, that means only 'latest' should have. alpine has that and iirc that works fine, so indeed just one postgresql-libs should be ok. but i'm not sure what to call it, because postgresql-libs-devel is weird, and -devel against the libs (latest only) is separate from -devel for the db itself (used to build out of tree postgres plugins, not latest only). what should we call them?

Reply:
easy solution is to actually keep this lib splitting as is, with libpq-devel/libecpg-devel, i.e. ignoring what i said. that seems ok

Reply:
and the todo pgtypes should stay in ecpg so that is done

Reply:
for things latest only, you can do

```
@subpackage("libecpg-devel", _is_latest)
```

and put `_is_latest = True` somewhere. though to get this to work like in alpine you also need `external-libpq.patch` that will only start getting applied on the old version while pulling new libpq in makedep. but let's leave everything i wrote in this comment for later